src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -787,6 +801,28 @@ void primitive_inst::do_runtime_skip_reorder() {
     }
 }
 
+void primitive_inst::do_runtime_skip_assign() {
+    OV_ITT_SCOPED_TASK(ov::intel_gpu::itt::domains::intel_gpu_plugin, openvino::itt::handle("do_runtime_skip_assign: " + id()));
+
+    if (can_be_optimized())
+        return;
+
+    if (get_node().is_type<concatenation>()) {
+        for (auto u : get_user_insts()) {
+            if (u->get_node().is_type<assign>()) {
```

Comment:
Is there any reason that we should check can_be_optimized of assing at the execution of concat?

Reply:
It is necessary to check reallocation of concatenation and determine can_be_optimized of assign at the same time.